ViewModel/Saving/PersistenceManagers/IASPersistenceManager.cs

```diff
@@ -116,8 +116,9 @@ public void Remove(ChildElement element)
         {
             if (element is IASElementSet)
             {
-                RemoveFromParentTable(element, TableName);
+                //remove from the cache first while you can still get the element's id.
                 StudyCacheForSaving.RemoveElement((IASElementSet)element);
```

Comment:
this can be simplified using pattern matching.

Reply:
done